CourseGrab/Home/HomeViewController.swift

```diff
@@ -97,8 +98,65 @@ class HomeViewController: UITableViewController {
         tableView.register(HomeTableViewHeader.self,
         forHeaderFooterViewReuseIdentifier: homeHeaderReuseId)
         tableView.register(HomeTableViewCell.self, forCellReuseIdentifier: homeCellReuseId)
+
+        if (!UserDefaults.standard.get) {
+            displayPermissionModal()
+        }
     }
 
+    func displayPermissionModal() {
+        let controller = SPPermissions.dialog([.notification])
+        controller.titleText = "Get In Your Courses"
+        controller.footerText = "Push notifications enhance the CourseGrab experience."
+        controller.dataSource = self as SPPermissionsDataSource
+        controller.delegate = self as SPPermissionsDelegate
+        controller.present(on: self)
+    }
+
+}
+
+extension HomeViewController: SPPermissionsDataSource {
+
+    func configure(_ cell: SPPermissionTableViewCell, for permission: SPPermission) -> SPPermissionTableViewCell {
+        cell.permissionTitleLabel.text = "Push Notifications"
+        cell.permissionDescriptionLabel.text = "Let us notify you when a section opens!"
+        cell.button.allowTitle = "Allow"
+        cell.button.allowedTitle = "Allowed"
+        cell.iconView.color = .courseGrabGreen
+        cell.button.allowedBackgroundColor = .courseGrabGreen
+        cell.button.allowBackgroundColor = .courseGrabGreen
+        cell.button.allowTitleColor = .white
+        return cell
+    }
+}
+extension HomeViewController: SPPermissionsDelegate {
+
+    func didAllow(permission: SPPermission) {
+        UserDefaults.standard.set(true, forKey: Constants.UserDefaults.didPromptPermission)
```

Comment:
not sure if you guys use user defaults enough in the app but i would recommend considering using `SwiftyUserDefaults` (https://github.com/sunshinejr/SwiftyUserDefaults)